### PR TITLE
Automatically apply CRDs for agent

### LINKED ISSUE #1318: Agent controller should automatically apply CRDs
Rather than requiring user intervention the agent controllers should upgrade their CRDs on start up similar to how Wrangler does.

## Acceptance Criteria
- [x] Agent controller applies required CRDs to cluster

### PR DESCRIPTION
**Pull request by dbason**
Fixes #1318

### CODE REVIEW

File: apis/logging/v1beta1/collector_config_types.go [hunk 1]
```diff
@@ -59,6 +63,20 @@ type CollectorConfigList struct {
 	Items           []CollectorConfig `json:"items"`
 }
 
+func CollectorConfigCRD() (*crd.CRD, error) {
```
Comment (alexandreLamarre) [inline apis/logging/v1beta1/collector_config_types.go:66]: Can we add the monitoring agent types as well to the wrangler pattern?

Comment (dbason) [reply apis/logging/v1beta1/collector_config_types.go:66]: We can, they weren't in main when I opened this though.  If they've been merged I will rebase

Comment (alexandreLamarre) [reply apis/logging/v1beta1/collector_config_types.go:66]: yeah they are in main as of tuesday I believe

File: config/crd/kustomization.yaml [hunk 2]
```diff
@@ -27,17 +27,17 @@ resources:
 - bases/core.opni.io_loggingclusters.yaml
 - bases/core.opni.io_monitoringclusters.yaml
 - bases/core.opni.io_natsclusters.yaml
-- bases/core.opni.io_collectors.yaml
+#- bases/core.opni.io_collectors.yaml
```
Comment (alexandreLamarre) [inline config/crd/kustomization.yaml:30]: Can probably remove these

**Review by alexandreLamarre — APPROVED** [review]
LGTM

**Pull request merged**